src/SFA.DAS.EmployerApprenticeshipsService.Infrastructure/Pipeline/Features/Sections/WhitelistPipelineSection.cs

```diff
@@ -0,0 +1,81 @@
+﻿using SFA.DAS.EAS.Domain.Models.FeatureToggles;
+using SFA.DAS.EAS.Infrastructure.Caching;
+using SFA.DAS.EAS.Infrastructure.Services;
+using SFA.DAS.EAS.Infrastructure.Services.FeatureToggle;
+using SFA.DAS.NLog.Logger;
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EAS.Infrastructure.Pipeline.Features.Sections
+{
+    public class WhitelistPipelineSection :
```

Comment:
The handler should ideally be composed rather than inheriting its behaviour. It probably shouldn't know about the caching strategy either.